toolkits/Dockerfile.cuda

```
@@ -0,0 +1,131 @@
+         ${CUDA_DEPS}/usr/lib/python${PYV}/dist-packages/ 2>/dev/null || :
+
+RUN printenv | \
+    grep -E '(CUDA|_CUDA|CUDNN|CURAND|CUBLAS|CUSOLVER|CUFFT|CUTENSOR|NPP|NCCL|NVIDIA|TENSORRT|POLYGRAPHY|NSIGHT|TRANSFORMER_ENGINE|HPCX|OPENUCX|EFA|MOFED|AWS_OFI_NCCL|LD_LIBRARY_PATH|PATH)' \
+  | while IFS='=' read -r key val; do \
+      if [ "$key" = "PATH" ]; then \
+        printf 'export PATH="$PATH:%s"\n' "$val"; \
+      elif [ "$key" = "LD_LIBRARY_PATH" ]; then \
+        printf 'export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:%s"\n' "$val"; \
+      else \
+        printf 'export %s="%s"\n' "$key" "$val"; \
+      fi; \
+    done > /tmp/cuda_env.sh
+RUN echo 'export NVIDIA_VISIBLE_DEVICES=all' >> /tmp/cuda_env.sh \
+  && echo 'export NVIDIA_DRIVER_CAPABILITIES=compute,utility' >> /tmp/cuda_env.sh
+
+RUN chown -R 1000:1000 /tmp/cuda_env.sh
+RUN chmod +x /tmp/cuda_env.sh
+
+FROM ros:${ROS_DISTRO}-ros-base AS environment-variables
+
+ENV ROS_WS=/ros_ws
+RUN mkdir -p ${ROS_WS}/src
+WORKDIR ${ROS_WS}/src
+
+RUN bash -c "source /opt/ros/${ROS_DISTRO}/setup.bash && \
+  ros2 pkg create --build-type ament_cmake environment_variables"
+
+# package.xml
+RUN cat <<EOF > environment_variables/package.xml
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>environment_variables</name>
+  <version>0.0.0</version>
+  <description>Package to source hooks/scripts</description>
+  <maintainer email="support@aica.tech">root</maintainer>
+  <license>TODO: License declaration</license>
+
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
+  <test_depend>ament_lint_auto</test_depend>
+  <test_depend>ament_lint_common</test_depend>
+
+  <export>
+    <build_type>ament_cmake</build_type>
+  </export>
+</package>
+EOF
+
+# CMakeLists.txt
+RUN sed -i \
+  -e '/ament_package()/i ament_environment_hooks("${CMAKE_CURRENT_SOURCE_DIR}/hooks/00-environment-variables.sh")' \
+  -e '/ament_package()/i install(DIRECTORY scripts/ DESTINATION share/\${PROJECT_NAME}/scripts  FILES_MATCHING PATTERN "*.sh")' \
+  environment_variables/CMakeLists.txt
+
+# hook
+RUN mkdir -p environment_variables/hooks/ environment_variables/scripts/
+RUN cat <<EOF > environment_variables/hooks/00-environment-variables.sh
+#!/bin/bash
+ENV_SCRIPT_PATH=\$(readlink -f "\${BASH_SOURCE[0]}")
+PARENT="\$(dirname "\$ENV_SCRIPT_PATH")"
+SCRIPT_DIR="\${PARENT%/*}"
+for script in "\$SCRIPT_DIR"/scripts/*.sh; do
+  source "\$script"
+done
+EOF
+RUN chmod +x environment_variables/hooks/00-environment-variables.sh
+
+# colcon
+RUN cat <<EOF > environment_variables/colcon.pkg
+{
+  "hooks": ["share/environment_variables/environment/00-environment-variables.sh"]
+}
+EOF
+
+WORKDIR $ROS_WS
+RUN bash -c "source /opt/ros/${ROS_DISTRO}/setup.bash && \
+  colcon build"
+RUN cp -r $ROS_WS/install/environment_variables /environment_variables
+
+FROM scratch
```

Nitpick but would prefer that each stage has a name, so maybe
```suggestion
FROM scratch AS final
```